lib/formats/html/html.flow

```diff
@@ -458,37 +458,42 @@ doUnescapeHtml(acc : string, s : string, numericOnly : bool) -> string {
 	}
 }
 
-urlEncode(s){
+urlEncodeMap = {
+	char2int = \c -> getCharCodeAt(c, 0);
 	// refers to http://www.december.com/html/spec/esccodes.html
-	r = \from,to -> \t -> strReplace(t, from, to);
-	s
-	|> r("%", "%25") // important to be first
-	|> r(" ", "%20")
-	|> r("\"", "%22") //needed for link be clickable
-	|> r("<", "%3C")
-	|> r(">", "%3E")
-	|> r("#", "%23")
-	|> r("{", "%7B")
-	|> r("}", "%7D")
-	|> r("|", "%7C")
-	|> r("\\", "%5C")
-	|> r("^", "%5E")
-	|> r("~","%7E")
-	|> r("[","%5B")
-	|> r("]","%5D")
-	|> r("`","%60")
-	|> r(";","%3B")
-	|> r("/","%2F")
-	|> r("?","%3F")
-	|> r(":","%3A")
-	|> r("@","%40")
-	|> r("=","%3D")
-	|> r("&","%26")
-	|> r("$","%24")
-	|> r("'","%27")
-	|> r("–","%E2%80%93")
-	|> r("™", "%E2%84%A2")
-	|> r("\n", "%0a")
+	pairs2tree([
+		Pair(char2int("%"), "%25"), // important to be first
```

Comment:
Notice the comment. It says that one has to be first. It can not be correct to use a tree where the order is changed. We have a multi-string replacement function, which uses an array. Why not just use that?

Reply:
ok, I'll correct it